wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -93,8 +95,25 @@ data Notification = BlockValidated Block
                   deriving (Show, Eq, Ord)
 
 -- manually records the list of transactions to be submitted
-newtype EmulatedWalletApi a = EmulatedWalletApi { runEmulatedWalletApi :: (ExceptT WalletAPIError (StateT WalletState (Writer [Tx] ))) a }
-    deriving (Functor, Applicative, Monad, MonadState WalletState, MonadWriter [Tx], MonadError WalletAPIError)
+newtype EmulatedWalletApi a = EmulatedWalletApi { runEmulatedWalletApi :: (ExceptT WalletAPIError (StateT WalletState (Writer (WalletLog, [Tx])))) a }
+    deriving (Functor, Applicative, Monad, MonadState WalletState, MonadError WalletAPIError)
+
+-- NB We need to write both [Tx] and WalletLog values, but
```

Comment:
I think I'd rather use the derived instance for the pair and then write `tellTx` and `tellLog` that go to the first or second respectively.

Reply:
but we need `MonadWriter WalletLog` in the WalletAPI actions

Reply:
Ugh, I guess. Or we could add another actual typeclass for that. It's fine.

Reply:
`MonadWriter [Tx]` is only needed to implement `submitTxn`, so I could just add a field to `WalletState` that keeps track of the transactions, and reserve the `MonadWriter` for log messages

Reply:
> Or we could add another actual typeclass for that

That works, too

Reply:
I don't really mind - I think any of those is fine.